Intersect.Client/Entities/Player.cs

```diff
@@ -2372,6 +2376,48 @@ public void DrawOverheadInfoOnHover()
             }
         }
 
+        public override bool ShouldDrawHpBar
+        {
+            get
+            {
+                if (LatestMap == default)
+                {
+                    return false;
+                }
+
+                if (!ShouldDraw)
+                {
+                    return false;
+                }
+
+                var health = Vital[(int)Vitals.Health];
+                if (health < 1)
+                {
+                    return false;
+                }
+
+                var maxHealth = MaxVital[(int)Vitals.Health];
+                var shieldSize = GetShieldSize();
+
+                var player = this;
+                bool isMe = player.Id == Globals.Me.Id,
+                    isFriend = !isMe && Globals.Me.IsFriend(player),
+                    isPartyMate = !isMe && Globals.Me.IsInMyParty(player),
+                    isGuildMate = !isMe && Globals.Me.IsGuildMate(player),
+                    friendSettingMatch = isFriend && Globals.Database.FriendOverheadHpBar,
+                    guildSettingMatch = isGuildMate && Globals.Database.GuildMemberOverheadHpBar;
+
+                return (shieldSize > 0 || health != maxHealth) ||
+                       (isMe && Globals.Database.MyOverheadHpBar) ||
+                       (!isMe && !isFriend && !isGuildMate && !isPartyMate && Globals.Database.PlayerOverheadHpBar) ||
+                       (isPartyMate && Globals.Database.PartyMemberOverheadHpBar) ||
+                       (friendSettingMatch && guildSettingMatch && !isPartyMate) ||
+                       (friendSettingMatch && !guildSettingMatch && !isPartyMate) ||
+                       (guildSettingMatch && !friendSettingMatch && !isPartyMate) ||
+                       IsHovered;
```

Comment:
The rest of this is wildly inefficient (it makes _every_ expensive check possible) and very illegible (it's difficult to separate anything).

Just break each of the other options into separate if statements, prioritizing the cheapest calls.

```csharp

if (IsHovered) {
    return true;
}

if (player == Globals.Me)
{
    return Globals.Database.MyOverheadHpBar;
}

if (<show party overhead> && <is in party>)
{
    return true;
}

if (<show friend overhead> && <is friend>)
{
    return true;
}

if (<show guild overhead> && <is in guild>)
{
    return true;
}

return general player overhead;
```